IdentityCore/src/MSIDCache.m

```diff
@@ -64,21 +64,21 @@ - (id)objectForKey:(id)key
 
 - (void)setObject:(id)obj forKey:(id)key
 {
-    dispatch_barrier_sync(self.synchronizationQueue, ^{
+    dispatch_barrier_async(self.synchronizationQueue, ^{
```

Comment:
Don't we want to wait for the operation to complete? e.g. what if the caller executes this and is reading cache right after executing and is expecting the cache to be in certain state it left it?

Reply:
When you use dispatch_barrier, it waits for all operations before it (eg read) to finish first and then executes exclusively. I forwarded a link on your slack which has more info on it. Ideally , the MSIDCache should also be a singleton, so multiple threads should not be able to create multiple instances of the dictionary. But our current constructor takes in dictionary as parameter so I could not make it a singleton.

Reply:
Yes, I know it will wait for operations to finish before executing barrier. However, according to Apple documents there's a difference and I want to make sure we consider all scenarios before changing this.

```
dispatch_barrier_sync
Submits a barrier block to a dispatch queue for synchronous execution. Unlike dispatch_barrier_async, this function does not return until the barrier block has finished. Calling this function and targeting the current queue results in deadlock.
When the barrier block reaches the front of a private concurrent queue, it is not executed immediately. Instead, the queue waits until its currently executing blocks finish executing. At that point, the queue executes the barrier block by itself. Any blocks submitted after the barrier block are not executed until the barrier block completes.
The queue you specify should be a concurrent queue that you create yourself using the dispatch_queue_create function. If the queue you pass to this function is a serial queue or one of the global concurrent queues, this function behaves like the dispatch_sync function.
Unlike with dispatch_barrier_async, no retain is performed on the target queue. Because calls to this function are synchronous, it "borrows" the reference of the caller. Moreover, no Block_copy is performed on the block.
As an optimization, this function invokes the barrier block on the current thread when possible.
```